File: server/scripts/checkSavedData.py
import openpyxl
import os

file_path = r'c:\Users\aniam\Desktop\PROJET TECHWEB\server\data\BDD1.xlsx'

if not os.path.exists(file_path):
    print("File not found at:", file_path)
else:
    print("File found! Loading...")
    wb = openpyxl.load_workbook(file_path)
    ws = wb['Utisers']
    
    # Find user with ID 1769718198386
    user_id = '1769718198386'
    found = False
    
    for row in ws.iter_rows(min_row=2, values_only=False):
        if str(row[0].value) == user_id:
            found = True
            print(f"\n✓ User found! Here's the saved data:\n")
            print(f"ID: {row[0].value}")
            print(f"Prénom: {row[1].value}")
            print(f"Nom: {row[2].value}")
            print(f"Email: {row[3].value}")
            print(f"Mot de passe (hash): {row[4].value}")
            print(f"Téléphone (numTel): {row[5].value}")
            print(f"Langue: {row[6].value}")
            print(f"Thème: {row[7].value}")
            print(f"Date création: {row[8].value}")
            print(f"Rôle: {row[9].value}")
            print(f"Adresse: {row[10].value}")
            print(f"Ville: {row[11].value}")
            print(f"Code postal: {row[12].value}")
            print(f"Pays: {row[13].value}")
            print(f"Nom carte: {row[14].value}")
            print(f"Numéro carte: {row[15].value}")
            print(f"Date expiration: {row[16].value}")
            print(f"CVV: {row[17].value}")
            break
    
    if not found:
        print(f"\n✗ User with ID {user_id} not found in database")
        print("\nFirst 5 users in database:")
        count = 0
        for row in ws.iter_rows(min_row=2, values_only=False):
            if count >= 5:
                break
            print(f"ID: {row[0].value}, Email: {row[3].value}")
            count += 1

File: server/scripts/checkSavedData.js
const ExcelJS = require('exceljs');
const path = require('path');

async function checkSavedData() {
  const filePath = path.join(__dirname, './data/BDD1.xlsx');
  const workbook = new ExcelJS.Workbook();
  await workbook.xlsx.readFile(filePath);
  const sheet = workbook.getWorksheet('Utisers');
  
  // Un essayer pour un id specifique, prck n L API marche pas 
  const userId = '1769718198386';
  let foundUser = false;
  
  sheet.eachRow((row, rowNumber) => {
    if (rowNumber === 1) return; 
    const id = String(row.getCell(1).value || '');
    
    if (id === userId) {
      foundUser = true;
      console.log('✓ User found in database!');
      console.log('\nSaved data:');
      console.log('ID:', row.getCell(1).value);
      console.log('Prénom:', row.getCell(2).value);
      console.log('Nom:', row.getCell(3).value);
      console.log('Email:', row.getCell(4).value);
      console.log('Mot de passe (hash):', row.getCell(5).value);
      console.log('Téléphone (numTel):', row.getCell(6).value);
      console.log('Langue:', row.getCell(7).value);
      console.log('Thème:', row.getCell(8).value);
      console.log('Date création:', row.getCell(9).value);
      console.log('Rôle:', row.getCell(10).value);
      console.log('Adresse:', row.getCell(11).value);
      console.log('Ville:', row.getCell(12).value);
      console.log('Code postal:', row.getCell(13).value);
      console.log('Pays:', row.getCell(14).value);
      console.log('Nom carte:', row.getCell(15).value);
      console.log('Numéro carte:', row.getCell(16).value);
      console.log('Date expiration:', row.getCell(17).value);
      console.log('CVV:', row.getCell(18).value);
    }
  });
  
  if (!foundUser) {
    console.log('✗ User with ID', userId, 'not found in database');
  }
}

checkSavedData().catch(err => console.error('Error:', err));
